x-pack/plugins/security/server/audit/audit_service.ts

```diff
@@ -250,10 +241,10 @@ export function filterEvent(
   if (ignoreFilters) {
     return !ignoreFilters.some(
       (rule) =>
-        (!rule.actions || rule.actions.includes(event.event.action)) &&
-        (!rule.categories || rule.categories.includes(event.event.category!)) &&
-        (!rule.types || rule.types.includes(event.event.type!)) &&
-        (!rule.outcomes || rule.outcomes.includes(event.event.outcome!)) &&
+        (!rule.actions || rule.actions.includes(event.event?.action!)) &&
+        (!rule.categories || event.event?.category?.every((c) => rule.categories?.includes(c))) &&
+        (!rule.types || event.event?.type?.every((t) => rule.types?.includes(t))) &&
```

Comment:
I think it should be enough if one of the categories/types matches.

Say we have the following ignore filter:
```
- categories: [user]
```

and are filtering the following event:
```
{
  category: [authentication, user]
}
```

Would we expect the event to be filtered out or be logged? 

Either way it would be good to document how this will work and add a couple of unit tests.

Reply:
tbh I found the filters difficult to reason about, so I appreciate you bringing this up.

> Would we expect the event to be filtered out or be logged?

I would expect that filtering takes the most conservative approach possible so that we don't accidentally filter something we shouldn't.

My changes here assume that a provided ignore filter would only apply if it matched _all_ items in the event, i.e.:

```
- categories: [web]
```

Would filter out
```
{
  category: [web]
}
```

But not
```
{
  category: [authentication, web]
}
```

And:
```
- categories: [authentication, web]
```

Would filter out
```
{
  category: [authentication, web]
}
```

And also
```
{
  category: [web]
}
```

But not
```
{
  category: [configuration]
}
```

This is also consistent with the way we treat each of the different fields... if one criteria doesn't match, we keep the whole event instead of filtering out: https://github.com/lukeelmers/kibana/blob/fb3e7eb9258b5dcdc61ee5d0c3b3b8a4f616756b/x-pack/plugins/security/server/audit/audit_service.test.ts#L379 

I've added some more unit tests & a comment explaining this, but LMK if you think we should handle differently.

Reply:
I agree with this, thanks for clarifying different scenarios.